muspy/classes.py

```diff
@@ -331,10 +331,38 @@ def _validate(self, attr: str, recursive: bool):
     def _adjust_time(
         self, func: Callable[[int], int], attr: str, recursive: bool
     ):
-        if attr == "time":
-            self.time = func(self.time)
-        elif attr == "duration":
-            self.duration = func(self.duration)
+        raise NotImplementedError
+
+    def adjust_time(
+        self,
+        func: Callable[[int], int],
+        attr: Optional[str] = None,
+        recursive: bool = True,
+    ) -> "Note":
+        """Adjust the timing of the note.
+
+        Parameters
+        ----------
+        func : callable
+            The function used to compute the new timing from the old
+            timing, i.e., `new_time = func(old_time)`.
+        attr : str
+            Attribute to adjust. Defaults to adjust all attributes.
+        recursive : bool
+            Whether to apply recursively. Defaults to True.
+
+        Returns
+        -------
+        Object itself.
+
+        """
+        if attr is not None and attr != "time":
+            raise AttributeError(f"'Note' object has no attribute '{attr}'")
```

Comment:
This might be confusing since there are other attributes than "time", we just don't support adjusting them.

Reply:
Oops. You are right. I will fix this later.

In fact, several methods of Base do not handle `AttributeError` properly.